File: android/src/main/kotlin/com/example/easy_video_editor/handler/MaxFpsCommand.kt
package com.example.easy_video_editor.handler

import android.content.Context
import androidx.media3.common.util.UnstableApi
import com.example.easy_video_editor.command.Command
import com.example.easy_video_editor.utils.OperationManager
import com.example.easy_video_editor.utils.VideoUtils
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.Job
import kotlinx.coroutines.cancel
import kotlinx.coroutines.launch

class MaxFpsCommand(private val context: Context) : Command {
    @UnstableApi
    override fun execute(call: MethodCall, result: MethodChannel.Result) {
        val videoPath = call.argument<String>("videoPath")
        val maxFps = call.argument<Number>("maxFps")?.toInt()

        if (videoPath == null || maxFps == null) {
            result.error(
                "INVALID_ARGUMENTS",
                "Missing required arguments: videoPath or maxFps",
                null
            )
            return
        }

        // Create a new scope that's tied only to this method call
        val methodScope = CoroutineScope(Dispatchers.Main + Job())

        // Register with operation manager for cancellation support
        val operationId = OperationManager.generateOperationId()
        OperationManager.registerOperation(operationId, methodScope)
        
        methodScope.launch {
            try {
                val outputPath = VideoUtils.setMaxFps(
                    context = context,
                    videoPath = videoPath,
                    maxFps = maxFps
                )
                result.success(outputPath)
            } catch (e: Exception) {
                result.error("MAX_FPS_ERROR", e.message, null)
            } finally {
                OperationManager.cancelOperation(operationId)
                methodScope.cancel()
            }
        }
    }
} 

File: ios/Classes/handler/MaxFpsCommand.swift
import Flutter
import Foundation
import AVFoundation

class MaxFpsCommand: Command {
    func execute(call: FlutterMethodCall, result: @escaping FlutterResult) {
        guard let args = call.arguments as? [String: Any],
              let videoPath = args["videoPath"] as? String,
              let maxFps = args["maxFps"] as? Int else {
            result(FlutterError(code: "INVALID_ARGUMENTS",
                              message: "Missing required arguments: videoPath or maxFps",
                              details: nil))
            return
        }
        
        // Create a new operation ID for cancellation support
        let operationId = OperationManager.shared.generateOperationId()
        
        // Create a dispatch queue for video processing
        let queue = DispatchQueue(label: "com.example.easy_video_editor.maxFps", qos: .userInitiated)
        
        queue.async {
            do {
                let outputPath = try VideoUtils.setMaxFps(videoPath: videoPath, maxFps: maxFps)
                DispatchQueue.main.async {
                    result(outputPath)
                }
            } catch let error as VideoError {
                DispatchQueue.main.async {
                    switch error {
                    case .fileNotFound:
                        result(FlutterError(code: "FILE_NOT_FOUND",
                                          message: "Video file not found",
                                          details: nil))
                    case .invalidParameters:
                        result(FlutterError(code: "INVALID_PARAMETERS",
                                          message: "Invalid parameters provided",
                                          details: nil))
                    case .exportFailed(let message):
                        result(FlutterError(code: "EXPORT_FAILED",
                                          message: message,
                                          details: nil))
                    case .invalidAsset:
                        result(FlutterError(code: "INVALID_ASSET",
                                          message: "Invalid video asset",
                                          details: nil))
                    default:
                        result(FlutterError(code: "MAX_FPS_ERROR",
                                          message: error.localizedDescription,
                                          details: nil))
                    }
                }
            } catch {
                DispatchQueue.main.async {
                    result(FlutterError(code: "MAX_FPS_ERROR",
                                      message: error.localizedDescription,
                                      details: nil))
                }
            }
            
            // Clean up operation
            OperationManager.shared.cancelOperation(operationId)
        }
    }
} 